packages/devtools_app/test/flutter/network_profiler_test.dart

```diff
@@ -0,0 +1,302 @@
+// setHttpTimelineLogging RPCs.
+Future<void> clearTimeouts(WidgetTester tester) async =>
+    await tester.pumpAndSettle(const Duration(seconds: 5));
+
+void main() {
+  NetworkController controller;
+  FakeServiceManager fakeServiceManager;
+  Timeline timeline;
+
+  const windowSize = Size(1599.0, 1000.0);
+
+  setUpAll(() async => timeline = await loadNetworkProfileTimeline());
+
+  group('Network Profiler', () {
+    setUp(() async {
+      fakeServiceManager =
+          FakeServiceManager(useFakeService: true, timelineData: timeline);
+      (fakeServiceManager.service as FakeVmService)
+          .getHttpEnableTimelineLoggingResult = false;
+      setGlobal(ServiceConnectionManager, fakeServiceManager);
+    });
+
+    testWidgets('builds its tab', (WidgetTester tester) async {
+      await tester.pumpWidget(wrap(
+        Builder(builder: const NetworkScreen().buildTab),
+      ));
+      expect(find.text('Network'), findsOneWidget);
+    });
+
+    testWidgetsWithWindowSize('starts and stops', windowSize, (
+      WidgetTester tester,
+    ) async {
+      controller = await pumpNetworkScreen(tester);
+
+      // Ensure we're not recording initially.
+      expect(controller.isPolling, false);
+      expect(controller.recordingNotifier.value, false);
+
+      // Start recording.
+      await tester.tap(find.byKey(NetworkScreen.recordButtonKey));
+      await tester.pump();
+
+      // Check that we're polling.
+      expect(controller.isPolling, true);
+      expect(controller.recordingNotifier.value, true);
+
+      // Stop recording.
+      await tester.tap(find.byKey(NetworkScreen.pauseButtonKey));
+      await tester.pump();
+
+      // Check that we've stopped polling.
+      expect(controller.isPolling, false);
+      expect(controller.recordingNotifier.value, false);
+
+      await clearTimeouts(tester);
+    });
+
+    Future<void> loadRequestsAndCheck(WidgetTester tester) async {
+      final splitFinder = find.byType(Split);
+
+      // We're not recording; only expect the instructions and buttons to be
+      // visible.
+      expect(splitFinder, findsNothing);
+      expect(find.byKey(NetworkScreen.clearButtonKey), findsOneWidget);
+      expect(find.byKey(NetworkScreen.recordButtonKey), findsOneWidget);
+      expect(find.byKey(NetworkScreen.pauseButtonKey), findsOneWidget);
+      expect(
+        find.byKey(NetworkScreen.recordingInstructionsKey),
+        findsOneWidget,
+      );
+
+      // Start recording but don't advance the clock in order to check the
+      // loading spinner is displayed before requests are populated.
+      await tester.tap(find.byKey(NetworkScreen.recordButtonKey));
+      await tester.pump();
+
+      expect(splitFinder, findsOneWidget);
+      expect(find.byType(CircularProgressIndicator), findsOneWidget);
+
+      // Advance the clock to populate the HTTP requests table.
+      await tester.pump(const Duration(seconds: 1));
```

Comment:
can we wait for an event instead of sleeping for a second? flaky tests can be a major problem when tests depend on timing on travis.

Reply:
This isn't a true "sleep" AFAIK. `flutter test` uses fake async (see [this article](https://medium.com/flutter/understanding-async-in-flutter-tests-a304a7604b3c) for a description), so this isn't an actual delay but makes all time related code act as if a second has passed.